### PR TITLE
1k+ assets (part 1)

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

This PR introduces `PersistedAccountData` data type that is `AccountData` + `ExtendedAssetHoldings`. It will be a blob stored into `accountbase` table instead of plain `AccountData`
Two abbreviations introduced: **pad** is for `PersistedAccountData` and **dbad** is for `DbAccountData` = `PersistedAccountData` + **rowId**.

### CODE REVIEW

File: ledger/ledgercore/persistedacctdata.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	// assetId3 = AmountsAssetIndicesOffsets[2] + assetIdx2
+	AssetOffsets []basics.AssetIndex `codec:"ao,allocbound=MaxHoldingGroupSize"`
+
+	// Holding amount
+	// same number of elements as in AmountsAssetIndicesOffsets
+	Amounts []uint64 `codec:"a,allocbound=MaxHoldingGroupSize"`
+
+	// Holding "frozen" flag
+	// same number of elements as in AmountsAssetIndicesOffsets
+	Frozens []bool `codec:"f,allocbound=MaxHoldingGroupSize"`
```
Comment (tsachiherman) [inline ledger/ledgercore/persistedacctdata.go:69]: Would this be the most efficient way to store array of bits ? What if we would use byte array for this one, and encode the bits in a little/big endian fashion ?

File: ledger/ledgercore/persistedacctdata.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+// ExtendedAssetHolding is AccountData's extension for storing asset holdings
+type ExtendedAssetHolding struct {
+	_struct struct{}             `codec:",omitempty,omitemptyarray"`
```
Comment (tsachiherman) [inline ledger/ledgercore/persistedacctdata.go:74]: nit: add empty line after `_struct` line

File: ledger/ledgercore/persistedacctdata.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	Count   uint32               `codec:"c"`
+	Groups  []AssetsHoldingGroup `codec:"gs,allocbound=4096"` // 1M asset holdings
```
Comment (tsachiherman) [inline ledger/ledgercore/persistedacctdata.go:76]: nit: define const for the 4096

File: ledger/ledgercore/persistedacctdata.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+	//msgp:ignore loaded
+	loaded bool
+}
+
+// PersistedAccountData represents actual data stored in DB
+type PersistedAccountData struct {
+	_struct struct{} `codec:",omitempty,omitemptyarray"`
```
Comment (tsachiherman) [inline ledger/ledgercore/persistedacctdata.go:84]: nit: empty line after `_struct`

File: ledger/ledgercore/persistedacctdata.go (hunk 3 above)
Comment (algorandskiy) [reply ledger/ledgercore/persistedacctdata.go:76]: done

File: ledger/ledgercore/persistedacctdata.go (hunk 4 above)
Comment (algorandskiy) [reply ledger/ledgercore/persistedacctdata.go:84]: done

File: ledger/ledgercore/persistedacctdata.go (hunk 2 above)
Comment (algorandskiy) [reply ledger/ledgercore/persistedacctdata.go:74]: done

File: ledger/ledgercore/persistedacctdata.go (hunk 1 above)
Comment (algorandskiy) [reply ledger/ledgercore/persistedacctdata.go:69]: Yes, makes sense, let's add it in subsequent PR

**Review by tsachiherman — APPROVED** [review]

**Pull request closed without merging**